flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/StreamingFileSink.java

```diff
@@ -251,14 +254,20 @@ T withBucketFactory(final BucketFactory<IN, BucketID> factory) {
 			return self();
 		}
 
+		@Internal
+		@Override
```

Comment:
@JingsongLi could you describe how this method is going to be used? Because in this commit, it seems that it is a private method of the builder :) Could it at least be package-private so that it is not exposed to the user?

Reply:
Sorry, let me explain the context:

In the file compaction, we need to write a file with the specified name. We hope to use `BucketWriter` to complete atomic file level writing.
The `BucketWriter` can complete atomic writing, but also shield the implementation details of BulkWriter, RowWiseWriter, OutputStreamBasedWriting and PathBasedWriting. That's great.

It's like the `createBuckets` method, We can not let it be package-private, because it is used by another module (flink-table).